src/lint_problems.ts

```diff
@@ -0,0 +1,279 @@
+  return "\n" + formattedLines.join('\n');
+}
+
+// Handle diagnostics for a single file
+export function handleGetFileLintProblems(actionData: ActionData): ActionResult {
+    if (!hasPermissions(PERMISSIONS.accessLintingAnalysis)) {
+        return actionResultNoPermission(PERMISSIONS.accessLintingAnalysis);
+    }
+    const relativePath = actionData?.params.file;
+    if (!relativePath) {
+        return actionResultMissingParameter('file');
+    }
+    const workspacePath = getWorkspacePath();
+    if (!workspacePath) {
+        logOutput("ERROR", 'Workspace folder not found.');
+        return actionResultFailure("Unable to get workspace path.");
+    }
+
+    try {
+        const absolutePath = path.join(workspacePath, relativePath);
+        const normalizedPath = normalizePath(absolutePath);
+        if (!isPathNeuroSafe(normalizedPath)) {
+            return actionResultNoAccess(normalizedPath);
+        }
+        if (!fs.existsSync(normalizedPath)) {
+            return actionResultFailure(`File '${relativePath}' does not exist.`);
+        }
+
+        const rawDiagnostics = vscode.languages.getDiagnostics(vscode.Uri.file(normalizedPath));
+        const formattedDiagnostics = getFormattedDiagnosticsForFile(relativePath, rawDiagnostics); // use relativePath
+        NEURO.client?.sendContext(`Linting problems for file ${relativePath}:${formattedDiagnostics}`);
+    } catch (error) {
+        NEURO.client?.sendContext(`Failed to get linting diagnostics for '${relativePath}'.`);
+        logOutput("ERROR", `Getting diagnostics for ${relativePath} failed: ${error}`);
+    }
+    return actionResultAccept();
+}
+
+
+// Handle diagnostics for a folder
+export function handleGetFolderLintProblems(actionData: ActionData): ActionResult {
+    if (!hasPermissions(PERMISSIONS.accessLintingAnalysis)) {
+        return actionResultNoPermission(PERMISSIONS.accessLintingAnalysis);
+    }
+    const relativeFolder = actionData?.params.folder;
+    if (!relativeFolder) {
+        return actionResultMissingParameter('folder');
+    }
+
+    const workspacePath = getWorkspacePath();
+    if (!workspacePath) {
+        logOutput("ERROR", 'Workspace folder not found.');
+        return actionResultFailure("Unable to get workspace path.");
+    }
+
+    try {
+        const absoluteFolderPath = path.join(workspacePath, relativeFolder);
+        const normalizedFolderPath = normalizePath(absoluteFolderPath);
+        if (!isPathNeuroSafe(normalizedFolderPath)) {
+            return actionResultNoAccess(normalizedFolderPath);
+        }
+        if (!fs.existsSync(normalizedFolderPath)) {
+            return actionResultFailure(`Folder '${relativeFolder}' does not exist.`);
+        }
+
+        const diagnostics = vscode.languages.getDiagnostics();
+        const folderDiagnostics = diagnostics.filter(([uri]) => {
+            return uri.fsPath.startsWith(normalizedFolderPath) && isPathNeuroSafe(uri.fsPath);
```

Comment:
Filter out files without problems
```suggestion
            return normalizePath(uri.fsPath).startsWith(normalizedFolderPath) && isPathNeuroSafe(uri.fsPath) && diags.length > 0;
```